janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLSubsetIterator.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.cql;
+
+import com.google.common.collect.Iterables;
+import org.janusgraph.diskstorage.StaticBuffer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+
+public class CQLSubsetIterator<TItem> implements Iterator<TItem> {
```

Comment:
(nitpick)
Codacy suggests:
```
Generics names should be a one letter long and upper case.
```

I usually use all upper case, but not always one letter.